Battery: add `not_charging_format` config option

Battery config
```toml
[[block]]
block = "battery"
format = "{percentage:2}"
device = "BAT0"
allow_missing = true
```

i3status-rust version: `0.21.2`
Linux kernel version: `5.15.17`

![image](https://user-images.githubusercontent.com/59267627/151661818-57b93424-c9ca-4d53-b1f4-6faa977490cb.png)

This is all I see
![image](https://user-images.githubusercontent.com/59267627/151661940-bfe9f2db-d7d3-4775-abf6-d4e9d21045ca.png)

Using `upower` also has the same behaviour.

Let me know if there anything I can do to debug this issue.

@MaxVerevkin 

Im getting the following compile error:
```
error: linking with `cc` failed: exit status: 1
...
  = note: /usr/bin/ld: cannot find -lnotmuch
          /usr/bin/ld: cannot find -lsensors
          collect2: error: ld returned 1 exit status
```

I am using rust 1.57 and cargo 0.58.0

--- a/src/blocks/battery.rs
+++ b/src/blocks/battery.rs
@@ -13,6 +13,7 @@
 //! `format` | A string to customise the output of this block. See below for available placeholders. | `" $icon $percentage "`
 //! `full_format` | Same as `format` but for when the battery is full | `" $icon "`
 //! `empty_format` | Same as `format` but for when the battery is empty | `" $icon "`
+//! `not_charging_format` | Same as `format` but for when the battery is not charging. Defaults to the full battery icon as many batteries report this status when they are full. | `" $icon "`
 //! `missing_format` | Same as `format` if the battery cannot be found. | `" $icon "`
 //! `info` | Minimum battery level, where state is set to info | `60`
 //! `good` | Minimum battery level, where state is set to good | `60`
@@ -83,6 +84,7 @@ pub struct Config {
     format: FormatConfig,
     full_format: FormatConfig,
     empty_format: FormatConfig,
+    not_charging_format: FormatConfig,
     missing_format: FormatConfig,
     #[default(60.0)]
     info: f64,
@@ -111,6 +113,7 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
     let format = config.format.with_default(" $icon $percentage ")?;
     let format_full = config.full_format.with_default(" $icon ")?;
     let format_empty = config.empty_format.with_default(" $icon ")?;
+    let format_not_charging = config.not_charging_format.with_default(" $icon ")?;
     let missing_format = config.missing_format.with_default(" $icon ")?;
     let mut widget = Widget::new();
 
@@ -136,7 +139,8 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
             Some(info) => {
                 widget.set_format(match info.status {
                     BatteryStatus::Empty => format_empty.clone(),
-                    BatteryStatus::Full | BatteryStatus::NotCharging => format_full.clone(),
+                    BatteryStatus::Full => format_full.clone(),
+                    BatteryStatus::NotCharging => format_not_charging.clone(),
                     _ => format.clone(),
                 });
 
@@ -159,7 +163,7 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
 
                 let (icon, state) = match (info.status, info.capacity) {
                     (BatteryStatus::Empty, _) => (battery_level_icon(0, false), State::Critical),
-                    (BatteryStatus::Full, _) => (battery_level_icon(100, false), State::Idle),
+                    (BatteryStatus::Full | BatteryStatus::NotCharging, _) => (battery_level_icon(100, false), State::Idle),
                     (status, capacity) => (
                         battery_level_icon(capacity as u8, status == BatteryStatus::Charging),
                         if status == BatteryStatus::Charging {

